include/swift/AST/DiagnosticsParse.def

```diff
@@ -619,8 +619,9 @@ ERROR(expected_type_function_result,PointsToFirstBadToken,
       "expected type for function result", ())
 ERROR(generic_non_function,PointsToFirstBadToken,
       "only syntactic function types can be generic", ())
-ERROR(rethrowing_function_type,PointsToFirstBadToken,
-      "only function declarations may be marked 'rethrows'", ())
+ERROR(rethrowing_function_type,none,
+      "only function declarations may be marked 'rethrows'; "
+      "did you mean 'throws'?", ())
```

Comment:
This doesn't make sense to me. The fact that `rethrows` isn't allowed in a function type (which I didn't realize until seeing this PR) seems like an unnecessary and surprising limitation. I would guess users will expect to be able to write `rethrows` function types, so suggesting `did you mean 'throws'?` will just be confusing.